packages/react-toolbar/Spec.md

```diff
@@ -76,10 +100,17 @@ type ToolbarDividerProps = ComponentProps<Partial<DividerSlots>>;
   <ToolbarDivider />
   <ToolbarButton />
   <ToolbarButton />
+  <ToolbarCheckbox />
+  <ToolbarRadioGroup>
```

Comment:
This is kind of unfortunate... It's good that you are reusing the checkbox and radio group components that we already have.

However, `Menu` uses a completely different selection API which controls all the selection from the top level
```
<Menu onCheckedValueChange={}>
  
</Menu>
```

We should make the `Toolbar` selection API consistent. The API that the `Menu` uses is actually better IMO because you don't need to control each individual checkbox and radio item.

```tsx
const [checkbox1, setCheckbox1] = useState(false)
const [checkbox1, setCheckbox1] = useState(false)
const [checkbox1, setCheckbox1] = useState(false)
const [checkbox1, setCheckbox1] = useState(false)

<Toolbar>

</Toolbar>
``` 

I guess

Reply:
@ling1726 I have pushed changes we talked about, have a look in latest commits

Reply:
@ling1726 I am actually surprised that menu has this API - I think it is better to not group the components state. It makes sense for radios, but checkoxes or toggles should ideally have atomic states I think. If the consumer wants to manage it as a group, they can do that on their side. 

For Toolbar, I think we want to reuse ToggleButton which has its own state anyway

Reply:
@jurokapsiar  when doing the research back during the early stages of menu, the libraries that offered selectable items as a core part of the menu would use a top level prop to control the selection state of items:
* reakit
* react spectrum
* ant design

It seemed to me that a lot of libraries offered menuitemcheckbox or menuitemradio as an afterthought. In hindsight I think I probably would have went with `MenuCheckboxGroup` or `MenuRadioGroup` but I would still force users to control checkboxes atomically, it's not convenient IMO